### PR TITLE
[firebase_remote_config] Add platform interface

### PR DESCRIPTION
**Pull request by balconywood**
## Description

Adds new `firebase_remote_config_platform_interface` package.

## Related Issues

FirebaseExtended/flutterfire#2889 FirebaseExtended/flutterfire#2115

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by kroikie** [comment]
Hi @balconywood thanks for the PR. Should you also include the functions to get the config values? Like [getString](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_remote_config/lib/src/remote_config.dart#L179) etc?

**Comment by balconywood** [comment]
> Hi @balconywood thanks for the PR. Should you also include the functions to get the config values? Like [getString](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_remote_config/lib/src/remote_config.dart#L179) etc?

That sounds like a good idea. I'll get working on it

File: packages/firebase_remote_config/firebase_remote_config_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## 1.0.0
```
[the remark this answers is not in the document]
Comment (balconywood) [reply packages/firebase_remote_config/firebase_remote_config_platform_interface/CHANGELOG.md:1]: Sounds good, I'll change this in the next commit.

File: packages/firebase_remote_config/firebase_remote_config_platform_interface/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!pubspec.yaml
```
[the remark this answers is not in the document]
Comment (balconywood) [reply packages/firebase_remote_config/firebase_remote_config_platform_interface/.gitignore:1]: I'll remove it in the next commit, thanks!

File: packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/enums.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of firebase_remote_config_platform_interface;
+
+/// ValueSource defines the possible sources of a config parameter value.
+enum ValueSource {
```
Comment (kroikie) [inline packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/enums.dart:8]: Nit: Could we add this to the `firebase_remote_config_platform_interface.dart` file?

**Comment by kroikie** [comment]
Also please sign the CLA

File: packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/enums.dart (hunk 3 above)
Comment (balconywood) [reply packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/enums.dart:8]: Sure, will do

**Comment by balconywood** [comment]
> Also please sign the CLA

I've signed the CLA, but think I accidentally used a placeholder email address in a few of my earlier commits, so the Googlebot doesn't recognize them as being authored by me, sorry about that. Should I do a rebase to correct the email address for the earlier commits?

File: packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/method_channel_firebase_remote_config.dart [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+          ValueSource.valueDefault,
+        );
+        _parameters[key] = remoteConfigValue;
+      }
+    });
+    return channel.invokeMethod<void>(
+        'RemoteConfig#setDefaults', <String, dynamic>{'defaults': defaults});
+  }
+
+  @override
```
[the remark this answers is not in the document]
Comment (kroikie) [reply packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/method_channel_firebase_remote_config.dart:69]: @balconywood @ditman is right. I thought those getters were calling out to the Method Channel sorry about the confusion there.

**Comment by Salakar** [comment]
Thank you for this PR. I am closing it however as the Remote Config plugin now already has a platform interface.

**Pull request closed without merging**